mmpose/datasets/datasets/hand/interhand2d_dataset.py

```diff
@@ -0,0 +1,269 @@
+import os
+from collections import OrderedDict
+
+import json_tricks as json
+import numpy as np
+
+from mmpose.datasets.builder import DATASETS
+from .hand_base_dataset import HandBaseDataset
+
+
+@DATASETS.register_module()
+class InterHand2DDataset(HandBaseDataset):
+    """InterHand2.6M 2D dataset for top-down hand pose estimation.
+
+    `InterHand2.6M: A Dataset and Baseline for 3D Interacting Hand Pose
+    Estimation from a Single RGB Image' Moon, Gyeongsik etal. ECCV'2020
+    More details can be found in the `paper
+    <https://arxiv.org/pdf/2008.09309.pdf>`__ .
+
+    The dataset loads raw features and apply specified transforms
+    to return a dict containing the image tensors and other information.
+
+    InterHand2.6M keypoint indexes::
+
+        0: 'thumb4',
+        1: 'thumb3',
+        2: 'thumb2',
+        3: 'thumb1',
+        4: 'forefinger4',
+        5: 'forefinger3',
+        6: 'forefinger2',
+        7: 'forefinger1',
+        8: 'middle_finger4',
+        9: 'middle_finger3',
+        10: 'middle_finger2',
+        11: 'middle_finger1',
+        12: 'ring_finger4',
+        13: 'ring_finger3',
+        14: 'ring_finger2',
+        15: 'ring_finger1',
+        16: 'pinky_finger4',
+        17: 'pinky_finger3',
+        18: 'pinky_finger2',
+        19: 'pinky_finger1',
+        20: 'wrist'
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        img_prefix (str): Path to a directory where images are held.
+            Default: None.
+        data_cfg (dict): config
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        test_mode (str): Store True when building test or
+            validation dataset. Default: False.
+    """
+
+    def __init__(self,
+                 ann_file,
+                 camera_file,
+                 joint_file,
+                 img_prefix,
+                 data_cfg,
+                 pipeline,
+                 test_mode=False):
+        super().__init__(
+            ann_file, img_prefix, data_cfg, pipeline, test_mode=test_mode)
+
+        self.ann_info['use_different_joint_weights'] = False
+        assert self.ann_info['num_joints'] == 21
+        self.ann_info['joint_weights'] = \
+            np.ones((self.ann_info['num_joints'], 1), dtype=np.float32)
+
+        self.dataset_name = 'interhand2d'
+        self.camera_file = camera_file
+        self.joint_file = joint_file
+        self.db = self._get_db()
+
+        print(f'=> num_images: {self.num_images}')
+        print(f'=> load {len(self.db)} samples')
+
+    def _cam2pixel_(self, cam_coord, f, c):
```

Comment:
`_cam2pixel_` -> `_cam2pixel`

the prefix `_` means internal function
the suffix `_` means inplace function, namely, it modifies arguments

same for `_world2cam_`